src/code/sys_cmpdma.c

```diff
@@ -19,14 +19,152 @@
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/sys_cmpdma/func_80178A94.s")
 
+typedef union {
+    struct {
+        u8 r;
+        u8 g;
+        u8 b;
+        u8 a;
+    };
+    s32 rgba;
+} Struct1_80178AC0;
+
+typedef union {
+    u16 a, b;
+    s32 ab;
+} Struct2_80178AC0;
+
+typedef struct {
+    /* 0x00 */ u32 data1;
+    /* 0x00 */ u32 data2;
```

Comment:
```suggestion
    /* 0x00 */ u32 start;
    /* 0x04 */ u32 end;
```
Based on usage, I think `start` and `end` would make sense.

Reply:
Can't say I agree.

Starting with`func_80178C80`, the function begins by reading in and storing into data1 the first word of a yaz archive, which stores the offset to the start of the data segment, relative to the file. Then in the refOff == 0 case, you read in `rom + 4` into data1, which stores the offset to the start of the second file relative to the _data segment_, which effectively is equivalent to the end of the first file/size of the first file. Only the last case do you have data1/data2 storing the start/end relative to the data segment. 

In func_80178E7C, I don't see how you could say that data1 is storing the start of anything; in the code it's being used to calculate how many times the loop should iterate for.

Reply:
Ah so it seems this just acts as a buffer to grab some kind of data, so yeah `start` and `end` don't make sense since the data is different. Not saying it would be better, but does just an array of 2 `u32`s work as well instead of defining a new struct?

Reply:
I've reworked this into a union